ThunderBasics/MultipleShadowView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  MultipleShadowView.swift
+//  ShadowView
+//
+//  Created by Simon Mitchell on 23/07/2020.
+//  Copyright © 2020 3 Sided Cube. All rights reserved.
+//
+
+import UIKit
+
+
+protocol CornerObservableLayerDelegate: class {
+    
+    func cornerObservableLayer(_ layer: CornerObservableLayer, didUpdateCornerRadius cornerRadius: CGFloat)
+    
+    func cornerObservableLayer(_ layer: CornerObservableLayer, didUpdateCornerCurve cornerCurve: CALayerCornerCurve)
+}
+
+internal class CornerObservableLayer: CALayer {
+    
+    /// Delegate for corner property changes
+    weak var cornerDelegate: CornerObservableLayerDelegate?
+    
+    override var cornerRadius: CGFloat {
+        didSet {
+            cornerDelegate?.cornerObservableLayer(self, didUpdateCornerRadius: cornerRadius)
+        }
+    }
+    
+    override var cornerCurve: CALayerCornerCurve {
+        didSet {
+            cornerDelegate?.cornerObservableLayer(self, didUpdateCornerCurve: cornerCurve)
+        }
+    }
+}
+
+/// A subclass of `UIView` which ensures multiple shadows have their
+/// bounds, and corner radius properties updated when the parent view's properties
+/// change.
+public class MultipleShadowView: UIView {
+    
+    public override class var layerClass: AnyClass {
+        return CornerObservableLayer.self
+    }
+    
+    /// Whether the shadow layers should update their corner radius to match the view's
+    /// layer's corner radius
+    public var matchShadowCornerRadius: Bool = true {
+        didSet {
+            guard matchShadowCornerRadius else { return }
```

Comment:
If the user sets this to `false` do we want to undo the corner radius of the `ShadowLayer`s

Reply:
I don't think so personally, I think when something is changing and you're responding to that change, you don't want to assume that the user wants to reset it when they're done responding to that change :) I'd expect the shadow corner radius to remain at it's final value :D Perhaps then we need a `setShadowCornerRadius` func or var?

Reply:
You can author it that way if you want :) If it respond and do something to turning it on I guess I'd expect it to also response and do something when you turn it off. But don't mind too much here. Can leave for now if you want.

Reply:
Snazzy, I think I'll leave for now and if we find ourselves being frustrated by this behaviour in the future we can change it!